pypit/arcoadd.py

```diff
@@ -0,0 +1,698 @@
+    scale_method : str, optional
+      Method for scaling
+       'auto' -- Use automatic method based on RMS of S/N
+       'hand' -- Use input scale factors
+       'median' -- Use calcualted median value
+    SN_MIN_MEDSCALE : float, optional
+      Maximum RMS S/N allowed to automatically apply median scaling
+    SN_MAX_MEDSCALE : float, optional
+      Maximum RMS S/N allowed to automatically apply median scaling
+
+    Returns
+    -------
+    scales : list of float or ndarray
+      Scale value (or arrays) applied to the data
+    omethod : str
+      Method applied (mainly useful if auto was adopted)
+       'hand'
+       'median'
+       'none_SN'
+    """
+    # Init
+    med_ref = None
+    rms_sn = np.sqrt(np.mean(sn2)) # Root Mean S/N**2 value for all spectra
+    # Check for wavelength registration
+    gdp = np.all(~spectra.data['flux'].mask, axis=0)
+    gidx = np.where(gdp)[0]
+    if not np.isclose(spectra.data['wave'][0,gidx[0]],spectra.data['wave'][1,gidx[0]]):
+        msgs.error("Input spectra are not registered!")
+    # Loop on exposures
+    scales = []
+    for qq in range(spectra.nspec):
+        if scale_method == 'hand':
+            omethod = 'hand'
+            # Input?
+            if hand_scale is None:
+                msgs.error("Need to provide hand_scale parameter, one value per spectrum")
+            spectra.data['flux'][qq,:] *= hand_scale[qq]
+            spectra.data['sig'][qq,:] /= hand_scale[qq]
+            #arrsky[*, j] = HAND_SCALE[j]*sclsky[*, j]
+            scales.append(hand_scale[qq])
+            #
+        elif ((rms_sn <= SN_MAX_MEDSCALE) and (rms_sn > SN_MIN_MEDSCALE)) or scale_method=='median':
+            omethod = 'median'
+            # Reference
+            if med_ref is None:
+                spectra.select = iref
+                med_ref, std_ref = median_flux(spectra)
+            # Calc
+            spectra.select = qq
+            med_spec, std_spec = median_flux(spectra)
+            # Apply
+            med_scale= np.minimum(med_ref/med_spec, 10.0)
+            spectra.data['flux'][qq,:] *= med_scale
+            spectra.data['sig'][qq,:] /= med_scale
+            #
+            scales.append(med_scale)
+        elif rms_sn <= SN_MIN_MEDSCALE:
+            omethod = 'none_SN'
+        elif (rms_sn > SN_MAX_MEDSCALE) or scale_method=='poly':
+            msgs.work("Should be using poly here, not median")
+            omethod = 'median'
+            # Reference
+            if med_ref is None:
+                spectra.select = iref
+                med_ref, std_ref = median_flux(spectra)
+            # Calc
+            spectra.select = qq
+            med_spec, std_spec = median_flux(spectra)
+            # Apply
+            med_scale= np.minimum(med_ref/med_spec, 10.0)
+            spectra.data['flux'][qq,:] *= med_scale
+            spectra.data['sig'][qq,:] /= med_scale
+            #
+            scales.append(med_scale)
+        else:
+            msgs.error('uh oh')
```

Comment:
I *think* the only way we could end up here is if a scale_method other than 'auto', 'hand', 'median', or 'poly' is input, so I've changed the error to reflect that, but do correct me if I'm wrong!